velox/exec/Task.cpp

```diff
@@ -651,15 +653,6 @@ RowVectorPtr Task::next(ContinueFuture* future) {
       planFragment_.executionStrategy,
       "Serial execution mode supports only ungrouped execution");
 
-  if (!splitsStates_.empty()) {
-    for (const auto& it : splitsStates_) {
-      VELOX_CHECK(
-          it.second.noMoreSplits,
```

Comment:
I assume we still need to assert that either noMoreSplits have been called or a barrier has been requested, no?

Reply:
Updated. Thanks for offline discussion